gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BaseTableType.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.bigquery.model.Table;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Base class for a Google BigQuery table type.
+ */
+public abstract class BaseTableType implements Serializable{
+
+  private static final long serialVersionUID = -374760330662959529L;
+
+  /**
+   * The table type.
+   */
+  public enum Type {
```

Comment:
I think changing the name from "Type" to something else would also save some confusion since schema and other subclass-related table properties are included.